presto-spi/src/main/java/com/facebook/presto/spi/memory/LocalMemoryContext.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.memory;
+
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.GuardedBy;
+import javax.annotation.concurrent.ThreadSafe;
+
+import static java.util.Objects.requireNonNull;
+
+@ThreadSafe
+public class LocalMemoryContext
+{
+    @GuardedBy("this")
+    private final AggregatedMemoryContext parentMemoryContext;
+    @Nullable
+    @GuardedBy("this")
+    private MemoryNotificationListener notificationListener;
+    @GuardedBy("this")
+    private long usedBytes;
+
+    public LocalMemoryContext(AggregatedMemoryContext parentMemoryContext)
+    {
+        this.parentMemoryContext = requireNonNull(parentMemoryContext);
+    }
+
+    public synchronized long getBytes()
+    {
+        return usedBytes;
+    }
+
+    public synchronized void setBytes(long bytes)
+    {
+        long oldLocalUsedBytes = this.usedBytes;
+        this.usedBytes = bytes;
+        parentMemoryContext.updateBytes(this.usedBytes - oldLocalUsedBytes);
+        if (notificationListener != null) {
+            notificationListener.memoryUsageUpdated(oldLocalUsedBytes, usedBytes);
+        }
+    }
+
+    public synchronized void addBytes(long bytes)
```

Comment:
The original design of memory context is so that callers do absolute (instead of relative) updates. It is a specific choice that LocalMemoryContext does not have a method like this.